internal/query/user.go

```diff
@@ -722,7 +723,7 @@
 		).ToSql()
 }
 
-func prepareUserQuery() (sq.SelectBuilder, func(*sql.Row) (*User, error)) {
+func prepareUserQuery(ctx context.Context, db prepareDatabase) (sq.SelectBuilder, func(*sql.Row) (*User, error)) {
```

Comment:
the as of system time clause has to appear in at least the top-level statement: https://www.cockroachlabs.com/docs/stable/as-of-system-time.html#using-as-of-system-time-in-subqueries

Reply:
It can occur multiple times if the duration is consistent.
Wouldn't it be less error prone to include it when we construct the queries with different functions?

Reply:
isn't adding it multiple times more error prone that adding it once?

Reply:
I'm a fan of less is more 😅